src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
 #endif
 }
 
+TEST_CASE("LegerTxnRoot prefetch soroban entries", "[ledgertxn]")
+{
+    auto ensureLedgerKeyMeterCanLoad =
+        [](std::shared_ptr<LedgerKeyMeter> lkMeter1,
+           std::vector<LedgerEntry> entries,
+           UnorderedSet<LedgerKey> expectedSuccessKeys,
+           UnorderedSet<LedgerKey> expectedFailedKeys) {
+            // Some, but not all keys in the expectedSuccessKeys set should
+            // be able to load. We expect that the number of keys that can
+            // load is less than the size of the expectedFailedKeys set.
+            int successfulFailures = 0;
+            for (const auto& entry : entries)
+            {
+                auto k = LedgerEntryKey(entry);
+                auto size = xdr::xdr_size(entry);
+                if (expectedSuccessKeys.find(k) != expectedSuccessKeys.end())
+                {
+                    REQUIRE(lkMeter1->canLoad(k, size));
+                }
+                else if (lkMeter1->canLoad(k, size))
+                {
+                    successfulFailures++;
+                }
+            }
+            REQUIRE(successfulFailures <= expectedFailedKeys.size());
+        };
+
+    auto runTest = [&](size_t cacheSize) {
+        Config cfg = getTestConfig();
+        cfg.ENTRY_CACHE_SIZE = cacheSize;
+
+        // Test setup.
+        VirtualClock clock;
+        cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+        Application::pointer app = createTestApplication(clock, cfg);
+        UnorderedSet<LedgerKey> keysToPrefetch;
+        auto& root = app->getLedgerTxnRoot();
+        LedgerTxn ltx(root);
+
+        UnorderedSet<LedgerKey> expectedSuccessKeys;
+        UnorderedSet<LedgerKey> expectedFailedKeys;
+        std::set<LedgerEntry> entrySet;
+        std::set<LedgerKey> deadKeySet;
+
+        auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
+        auto lkMeterHot = std::make_shared<LedgerKeyMeter>();
+
+        auto entries =
```

Comment:
Instead of generating random keys, I think we should just do the following:
```
auto contractDataEntry = generate...
auto classicEntry = generate...
auto TTL = buildTTLEntry(contractDataEntry) // See BucketListTests.cpp:847 for how to do this
auto dead = generateKeyOfTypeContractCode
```

That way, we can guarantee that we always generate a mix of classic and soroban keys. By building the TTL key instead of randomly generating it, we can have a more realistic test than if we populate footprints manually with TTLs, which is technically invalid.